components/brave_shields/browser/ad_block_service_browsertest.cc

```diff
@@ -744,3 +744,112 @@ IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CancelRequestOptionTest) {
   EXPECT_TRUE(as_expected);
   EXPECT_EQ(browser()->profile()->GetPrefs()->GetUint64(kAdsBlocked), 1ULL);
 }
+
+// Test simple cosmetic filtering
+IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CosmeticFilteringSimple) {
+  UpdateAdBlockInstanceWithRules(
+      "b.com###ad-banner\n"
+      "##.ad"
+  );
+
+  GURL tab_url = embedded_test_server()->GetURL("b.com",
+                                                "/cosmetic_filtering.html");
+  ui_test_utils::NavigateToURL(browser(), tab_url);
+
+  content::WebContents* contents =
+    browser()->tab_strip_model()->GetActiveWebContents();
+
+  bool as_expected = false;
+  ASSERT_TRUE(ExecuteScriptAndExtractBool(contents,
+                                          "checkSelector('#ad-banner', 'display', 'none')",
+                                          &as_expected));
+  EXPECT_TRUE(as_expected);
+
+  as_expected = false;
+  ASSERT_TRUE(ExecuteScriptAndExtractBool(contents,
+                                          "checkSelector('.ad-banner', 'display', 'block')",
+                                          &as_expected));
+  EXPECT_TRUE(as_expected);
+
+  as_expected = false;
+  ASSERT_TRUE(ExecuteScriptAndExtractBool(contents,
+                                          "checkSelector('.ad', 'display', 'none')",
+                                          &as_expected));
+  EXPECT_TRUE(as_expected);
+}
+
+// Test cosmetic filtering on elements added dynamically
+IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CosmeticFilteringDynamic) {
+  UpdateAdBlockInstanceWithRules(
```

Comment:
You might want a call to WaitForBraveExtensionAdded at the start above here.

Reply:
Hmm, that may help with https://github.com/brave/brave-core/pull/3303#discussion_r348114676... will try

Reply:
That doesn't seem to help, unfortunately.

Was there another reason you recommended adding that? It wasn't implemented on any of the other tests in the file either.